internal/shim-sgx/src/sgx_uarch.rs

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Contains microarchitecture structures of SGX. Keep the functionality
+//! minimal and stdlib free. Use the Intel SDM struct naming conventions,
+//! instead of Rust naming conventions.
+//!
+//! Chapter 34 of the Intel SDM contains detailed documentation for these
+//! structs.
+
+
+#[allow(non_camel_case_types)]
+
+/// Contains the fields that are part of the CMAC, signed with the SGX report
+/// key.
+#[allow(missing_docs)]
+#[derive(Clone, Copy)]
+#[repr(C)]
+pub struct REPORT_BODY {
+    pub cpusvn: [u64; 2],
+    pub miscselect: u32,
+    pub cet_attributes: u8,
+    reserved1: [u8; 12],
+    pub isvextnprodid: [u64; 2],
+    pub attributes: [u64; 2],
+    pub mrenclave: [u8; 32],
+    reserved2: [u8; 32],
+    pub mrsigner: [u8; 32],
+    reserved3: [u8; 32],
+    pub configid: [u8; 64],
+    pub isvprodid: u16,
+    pub isvsvn: u16,
+    pub configsvn: u16,
+    reserved4: [u8; 42],
+    pub isvfamilyid: [u64; 2],
+    pub reportdata: [u8; 64],
+}
+
+impl Default for REPORT_BODY {
+    fn default() -> Self {
+        REPORT_BODY {
+            cpusvn: [0, 0],
+            miscselect: 0,
+            cet_attributes: 0,
+            reserved1: [0; 12],
+            isvextnprodid: [0, 0],
+            attributes: [0, 0],
+            mrenclave: [0; 32],
+            reserved2: [0; 32],
+            mrsigner: [0; 32],
+            reserved3: [0; 32],
+            configid: [0; 64],
+            isvprodid: 0,
+            isvsvn: 0,
+            configsvn: 0,
+            reserved4: [0; 42],
+            isvfamilyid: [0, 0],
+            reportdata: [0; 64],
+        }
+    }
+}
+
+#[allow(missing_docs)]
```

Comment:
Please just provide some documentation.

Reply:
I can document the struct, but documenting separate fields does more harm than is a benefit.

Reply:
@jarkkojs Fine. Just put `#![allow(missing_docs)]` at the top of the file.

Reply:
I'm about to push with TI included.